### PR TITLE
Use typing in all classes

### PR DESCRIPTION
**Pull request by emymaria**
We should use typing everywhere and avoid any-Objects.

### CODE REVIEW

File: src/app/components/algorithms/algorithm-implementations-list/algorithm-implementations-list.component.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
 import { Component, Input, OnInit } from '@angular/core';
 import { AlgorithmService } from 'api-atlas/services/algorithm.service';
 import { ImplementationDto } from 'api-atlas/models/implementation-dto';
+import { PageImplementationDto } from 'api-atlas/models/page-implementation-dto';
```
Comment (salmma) [inline src/app/components/algorithms/algorithm-implementations-list/algorithm-implementations-list.component.ts:4]: Please remove unused imports as shown by the linter

File: src/app/components/algorithms/algorithm-related-algos-list/algorithm-related-algos-list.component.ts [hunk 2]
```diff
@@ -6,10 +6,14 @@ import { AlgorithmDto } from 'api-atlas/models/algorithm-dto';
 import { AlgorithmRelationTypeService } from 'api-atlas/services/algorithm-relation-type.service';
 import { Router } from '@angular/router';
 import { AlgorithmRelationTypeDto } from 'api-atlas/models';
+import { PageAlgorithmRelationDto } from 'api-atlas/models';
```
Comment (salmma) [inline src/app/components/algorithms/algorithm-related-algos-list/algorithm-related-algos-list.component.ts:9]: see comment above

File: src/app/components/algorithms/algorithm-related-patterns/algorithm-related-patterns.component.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { PatternRelationTypeService } from 'api-atlas/services/pattern-relation-
 import { AlgorithmService } from 'api-atlas/services/algorithm.service';
 import { PatternRelationTypeDto } from 'api-atlas/models/pattern-relation-type-dto';
 import { PatternRelationDto } from 'api-atlas/models';
+import { PagePatternRelationDto } from 'api-atlas/models';
```
Comment (salmma) [inline src/app/components/algorithms/algorithm-related-patterns/algorithm-related-patterns.component.ts:7]: see above

File: src/app/components/algorithms/implementation-view/implementation-publications-list/implementation-publications-list.component.ts [hunk 4]
```diff
@@ -98,6 +100,7 @@ export class ImplementationPublicationsListComponent implements OnInit {
   }
 
   updateDisplayedData(data): void {
+    console.log(data);
```
Comment (salmma) [inline src/app/components/algorithms/implementation-view/implementation-publications-list/implementation-publications-list.component.ts:103]: remove log

File: src/app/components/algorithms/implementation-view/implementation-view.component.ts [hunk 5]
```diff
@@ -122,15 +117,6 @@ export class ImplementationViewComponent implements OnInit {
     }
   }
 
-  onElementClicked(implementation: any): void {
```
Comment (salmma) [inline src/app/components/algorithms/implementation-view/implementation-view.component.ts:125]: this is not required?

File: src/app/components/application-areas/application-areas-list/application-areas-list.component.ts [hunk 6]
```diff
@@ -61,7 +65,7 @@ export class ApplicationAreasListComponent implements OnInit {
   }
 
   onAddElement(): void {
-    const params: any = {};
+    // On checking the references params has only body set and then used in line 83
```
Comment (salmma) [inline src/app/components/application-areas/application-areas-list/application-areas-list.component.ts:68]: referencing to a line is not helpful in the long run as code may change

Comment (salmma) [reply src/app/components/application-areas/application-areas-list/application-areas-list.component.ts:68]: in addition, I am not sure if the comment supports understanding - the code (as it is no complex code) _should_ be understandable enough such that it is clear what is done.

File: src/app/components/generics/dialogs/link-item-list-dialog.component.ts [hunk 7]
```diff
@@ -41,6 +43,6 @@ export interface DialogData {
   variableNames: string[];
   tableColumns: string[];
   pagingInfo?: any;
```
Comment (salmma) [inline src/app/components/generics/dialogs/link-item-list-dialog.component.ts:45]: here is an any

File: src/app/components/problem-types/problem-types-list/problem-types-list.component.ts [hunk 8]
```diff
@@ -19,8 +23,8 @@ export class ProblemTypesListComponent implements OnInit {
   problemTypes: any[] = [];
```
Comment (salmma) [inline src/app/components/problem-types/problem-types-list/problem-types-list.component.ts:23]: here is an any

**Review by salmma — CHANGES_REQUESTED** [review]
Please, see my comments.
Also see warnings of linter:
- remove unused imports
- still some anys left

File: src/app/components/algorithms/implementation-view/implementation-view.component.ts (hunk 5 above)
Comment (emymaria) [reply src/app/components/algorithms/implementation-view/implementation-view.component.ts:125]: No usages found

File: src/app/components/algorithms/algorithm-list/algorithm-list.component.ts [hunk 9]
```diff
@@ -67,7 +69,6 @@ export class AlgorithmListComponent implements OnInit {
   }
 
   onAddElement(): void {
-    const params: any = {};
     const dialogRef = this.utilService.createDialog(
       AddAlgorithmDialogComponent,
       {
```
Comment (salmma) [inline src/app/components/algorithms/algorithm-list/algorithm-list.component.ts:74]: in line 81: const algorithmDto: **any** = {
See linter warning

File: src/app/components/algorithms/algorithm-related-algos-list/algorithm-related-algos-list.component.ts [hunk 10]
```diff
@@ -27,8 +29,9 @@ export class AlgorithmRelatedAlgosListComponent implements OnInit {
     'description',
   ];
   tableColumns: string[] = ['Related Algorithm', 'Relation', 'Description'];
-  pagingInfo: any = {};
-  paginatorConfig: any = {
+  pagingInfo: PagingInfo<AlgorithmRelationDto> = {};
+
```
Comment (salmma) [inline src/app/components/algorithms/algorithm-related-algos-list/algorithm-related-algos-list.component.ts:33]: just remove the empty line :)

File: src/app/components/algorithms/algorithm-list/algorithm-list.component.ts (hunk 9 above)
Comment (emymaria) [reply src/app/components/algorithms/algorithm-list/algorithm-list.component.ts:74]: if (dialogResult) {
        const algorithmDto: any = {
          name: dialogResult.name,
          computationModel: dialogResult.computationModel,
        };

This is followed by this conditional check
        if (algorithmDto.computationModel === 'QUANTUM') {
          algorithmDto.quantumComputationModel =
            dialogResult.quantumComputationModel;
        }
So could not directly give the parameters inside the usage 
this.algorithmService.createAlgorithm({ body: algorithmDto }).subscribe(.........
as done in other files

Comment (salmma) [reply src/app/components/algorithms/algorithm-list/algorithm-list.component.ts:79]: see my last commit - I used a global variable wich is either `QuantumAlgorithmDto` or `ClassicAlgorithmDto` to remove the any

**Comment by emymaria** [comment]
**### List of remaining instances where type is specified as any**

1. **qc-atlas-ui\src\app\components\generics\chip-collection\chip-collection.component.ts**
 	From generics folder (11)
2. **qc-atlas-ui\src\app\components\generics\data-list\data-list.component.ts**
	From generics folder (18)
3. **qc-atlas-ui\src\app\components\generics\dialogs\confirm-dialog.component.ts**
 	From generics folder - Cannot remove this any since used for deleteion of different types of data (1)
4. **qc-atlas-ui\src\app\components\generics\dialogs\link-item-list-dialog.component.ts**
 	From generics folder - // Could not remove any type of selectedItems and pagingInfo(content?: T[]) used for different 
        types (3)
5. **qc-atlas-ui\src\app\components\generics\link-input\link-input.component.ts**
 	From generics folder (3)
6. **qc-atlas-ui\src\app\components\generics\navigation-breadcrumb\navigation-breadcrumb.component.ts**
 	EventEmitter<any>  (2)
7. **qc-atlas-ui\src\app\components\generics\property-input\abstract-value-accessor.ts**
 	From generics folder (8)
8. **qc-atlas-ui\src\app\http-interceptors\auth-interceptor.ts**
 	HttpEvent<any>(1) and HttpRequest<any> (4)
9. **qc-atlas-ui\src\app\util\show-on-feature.direct.ts**
	TemplateRef<any> line 18 (1)
10. **qc-atlas-ui\src\app\util\util.service.ts**
	Used in multiple components for diffrent data types (10)
 	line 67 method getColorOfSelectedButton() - No usages found 
        qc-atlas-ui\src\app\util\latex-renderer-service-constants.ts and qc-atlas-ui\src\app\util\util.service.ts
      line 50: The renderedData (response from util service) looks similiar to a blob but couldnot assign type Blob,So directly 
      assigned the response to latexBlob in utilService
      const latexBlob = this.latexRendererServiceConstants.createBlobFromRenderedResult(response); in line 200 replaced with 
       const latexBlob = response;

**Review by salmma — APPROVED** [review]
LG

**Pull request merged**